File: PatronInterprete/Introduccion_Interprete/src/Expresiones/Nativo.java
package Expresiones;


import Abstract.Instruccion;
import Simbolo.Arbol;
import Simbolo.Tipo;
import Simbolo.tablaSimbolos;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author JohnAlds
 */
public class Nativo extends Instruccion{
    public Object valor;

    public Nativo(Object valor, Tipo tipo, int linea, int col) {
        super(tipo, linea, col);
        this.valor = valor;
    }

    @Override
    public Object interpretar(Arbol arbol, tablaSimbolos tabla) {
        return this.valor;
    }
    
}

File: Jison/Expresiones/Nativo.ts
import { Instruccion } from "../Abstract/Instruccion"
import { Arbol } from "../Simbolo/Arbol"
import { TablaSimbolos } from "../Simbolo/TablaSimbolos"
import { Tipo } from "../Simbolo/Tipo"

export class Nativo extends Instruccion {

    public valor: any

    constructor(valor: any, tipo: Tipo, linea: number, columna: number) {
        super(tipo, linea, columna)
        this.valor = valor
    }

    interpretar(arbol: Arbol, tabla: TablaSimbolos): any {
        return this.valor
    }

}